File: sample-java/src/main/java/fr/bmartel/bboxapi/javasample/Summary.java
package fr.bmartel.bboxapi.javasample;

import com.github.kittinunf.fuel.core.FuelError;
import com.github.kittinunf.fuel.core.Handler;
import com.github.kittinunf.fuel.core.Request;
import com.github.kittinunf.fuel.core.Response;
import com.github.kittinunf.result.Result;
import fr.bmartel.bboxapi.BboxApi;
import kotlin.Triple;

import java.util.List;
import java.util.concurrent.CountDownLatch;

public class Summary {

    public static void main(String args[]) throws InterruptedException {
        BboxApi bboxapi = new BboxApi();

        //asynchronous call
        CountDownLatch latch = new CountDownLatch(1);
        bboxapi.getSummary(new Handler<List<fr.bmartel.bboxapi.model.Summary.Model>>() {
            @Override
            public void failure(Request request, Response response, FuelError error) {
                error.printStackTrace();
                latch.countDown();
            }

            @Override
            public void success(Request request, Response response, List<fr.bmartel.bboxapi.model.Summary.Model> data) {
                System.out.println(data);
                latch.countDown();
            }
        });
        latch.await();

        //synchronous call
        Triple<Request, Response, Result<List<fr.bmartel.bboxapi.model.Summary.Model>, FuelError>> data = bboxapi.getSummarySync();
        Request request = data.getFirst();
        Response response = data.getSecond();
        Result<List<fr.bmartel.bboxapi.model.Summary.Model>, FuelError> obj = data.getThird();
        System.out.println(obj.get());
    }
}

File: sample/src/main/kotlin/fr/bmartel/bboxapi/router/sample/Summary.kt
package fr.bmartel.bboxapi.router.sample

import com.github.kittinunf.fuel.core.HttpException
import com.github.kittinunf.result.Result
import fr.bmartel.bboxapi.router.BboxApi
import java.net.UnknownHostException
import java.util.concurrent.CountDownLatch

fun main(args: Array<String>) {
    val bboxapi = BboxApi()

    println("authentication attempts : ${bboxapi.attempts}")
    println("user is authenticated   : ${bboxapi.authenticated}")
    println("user is blocked         : ${bboxapi.blocked}")
    println("ban expiration date     : ${bboxapi.blockedUntil}")

    //asynchronous call
    val latch = CountDownLatch(1)
    bboxapi.getSummary { _, response, result ->
        when (result) {
            is Result.Failure -> {
                val ex = result.getException()
                when {
                    ex.exception is UnknownHostException -> println("hostname bbox.lan was not found")
                    ex.exception is HttpException -> println("http error : ${response.statusCode}")
                    else -> ex.printStackTrace()
                }
            }
            is Result.Success -> {
                val data = result.get()
                println(data)
            }
        }
        latch.countDown()
    }
    latch.await()

    //synchronous call
    val (_, _, result) = bboxapi.getSummarySync()
    when (result) {
        is Result.Failure -> {
            val ex = result.getException()
            println(ex)
        }
        is Result.Success -> {
            val data = result.get()
            println(data)
        }
    }
}